stage #2/module #3 Collections/lesson #2. List/list.md

```diff
@@ -0,0 +1,250 @@
+| int lastIndexOf(Object o)                            | Returns the index in this list of the last occurrence of the specified element, or -1 if the list does not contain this element.|
+| E remove(int index)                                  | Removes the element present at the specified position in the list.                                     |
+| Object[] toArray()                                   | Returns an array containing all of the elements in this list in the correct order.                     |
+| E set(int index, E element)                          | Replaces the specified element in the list, present at the specified position.                         |
+| boolean contains(Object o)                           | Returns true if the list contains the specified element                                               |
+| boolean containsAll(Collection<?> c)                 | Returns true if the list contains all the specified element                                           |
+| int indexOf(Object o)                                | Returns the index in this list of the first occurrence of the specified element, or -1 if the List does not contain this element.|
+| E remove(int index)                                  | Removes the element present at the specified position in the list.                                     |
+| boolean removeAll(Collection<?> c)                   | Removes all the elements from the list.                                                                |
+| void replaceAll(UnaryOperator<E> operator)           | Replaces all the elements from the list with the specified element.                                    |
+| void retainAll(Collection<?> c)                      | Retains all the elements in the list that are present in the specified collection.                     |
+| void sort(Comparator<? super E> c)                   | Sorts the elements of the list on the basis of specified comparator.                                   |
+| Spliterator<E> spliterator()                         | Creates spliterator over the elements in a list.                                                       |
+| int size()                                           | Returns the number of elements present in the list.                                                    |
+
+## Implementations:
+- AbstractList
+- AbstractSequentialList
+- ArrayList
+- LinkedList
+- CopyOnWriteArrayList
+- Stack
+- Vector
+
+## AbstractList
+The AbstractList class provides a partial implementation of the List interface (except get() and size() methods).
+
+## AbstractSequentialList
+The AbstractSequentialList class extends AbstractList to provide support for linked lists.
+
+![image info](media/AbstractList.png)
+
+## ArrayList
+
+### Overview
+
+The ArrayList class is a resizable array, which can be found in the java.util package. Java ArrayList class uses a dynamic array for storing the elements, has no size limit.

```

Comment:
Can you add a very tiny example with ArrayList (maybe copy past from above)?

Reply:
Thank you, done